src/supervisor-api.ts

```diff
@@ -176,22 +188,29 @@ export class SupervisorAPI {
 		apiTimeout: number,
 	): Promise<void> {
 		const localMode = await this.config.get('localMode');
-		await this.applyListeningRules(localMode || false, port, allowedInterfaces);
+		await this.applyRules(localMode || false, port, allowedInterfaces);
 
 		// Monitor the switching of local mode, and change which interfaces will
 		// be listened to based on that
 		this.config.on('change', changedConfig => {
 			if (changedConfig.localMode != null) {
-				this.applyListeningRules(
+				this.applyRules(
 					changedConfig.localMode || false,
 					port,
 					allowedInterfaces,
 				);
 			}
 		});
 
-		this.server = this.api.listen(port);
-		this.server.timeout = apiTimeout;
+		return new Promise(resolve => {
```

Comment:
the listen function offers a callback which wasn't being used. I refactored this to return a promise so code calling this can wait for the function to fully finish.